### PR TITLE
Fix layout issues and update markup

### LINKED ISSUE #4: Fix layout/ux issues
**User Statement:**
As a developer the layout should always appear the same across multiple browsers & operating systems

**Details:**

**Acceptance Criteria:**
Address the following comments from Louis

- [x] (compute capacity) wizard to short don’t see advanced button, need to scroll 
- [ ] (compute capacity) tooltip in advanced by debug options
- [ ] (compute capacity) confusing when the scrollbar isn’t visible. opened advanced and only saw vm endpoint settings for cpu, memory was below the fold
- [x] (storage capacity) inline hint text in file folder input
- [x] (storage capacity) in advanced section, hint text for “optional folder name
- [x] (networks) close space between diagram link and first network, should be able to see the advanced button
- [x] (networks) label font seems very small
- [x] (networks) advanced section - labels very small “container network”
- [x] (networks) alignment of http proxy, https proxy and dns server, port number should be on same line
- [x] (security) alignment of wrapped text in first bullet
- [x] (security) hint text in a couple places is pretty dark when in input
- [x] (security) when whitelist mode is enabled the input dropdown for secure is truncated as is the IP… input

### LINKED ISSUE #31: VCH wizard: Storage Capacity: File folder field placeholder truncated
The input field for the storage capacity file folder field truncates the placeholder text (see image) 
screen shot 2017-08-09 at 3 20 25 pm

### Steps to reproduce
1. Navigate to VIC plugin view
2. Open VCH tab view
3. Add a new VCH to open creation wizard
4. Proceed to step 3, Storage Capacity
5. The File folder input field truncates the placeholder text

### Expected behavior
The file folder field should be longer so that the placeholder text is seen

### LINKED ISSUE #32: add and delete icons positioning incorrect
The pattern to add or remove form-blocks or form fields in the creation wizard uses an add and delete icon.

The alignment and separation of these icons is incorrect in most places. An example is the addition or deletion of Container Networks in the Networks step of the wizard (see images)
screen shot 2017-08-09 at 3 25 47 pm
screen shot 2017-08-09 at 3 26 14 pm

### Expected positioning
The Add Icon should vertically align with the label of the form field and should have a 12px separation with the right side of the form field. Where the delete and add icon are shown, both should vertically align and have 12px separation.

Refer to the InVision mocks (see image)...
screen shot 2017-08-09 at 3 33 32 pm

### PR DESCRIPTION
**Pull request by cristianfalcone**
Fixes #4 
Fixes #31 
Fixes #32

### CODE REVIEW

File: ui/vic-ui-h5c/vic/src/vic-webapp/src/app/create-vch-wizard/storage-capacity/storage-capacity.html [hunk 1]
```diff
@@ -1,91 +1,194 @@
...
+        <div class="select form-control">
+          <select formControlName="baseImageSizeUnit">
+            <option value="MB">MB</option>
+            <option selected value="GB">GB</option>
+          </select>
         </div>
+
       </div>
+      <div class="col-xs-2"></div>
+      <div class="col-xs"></div>
```
Comment (jak-atx) [inline ui/vic-ui-h5c/vic/src/vic-webapp/src/app/create-vch-wizard/storage-capacity/storage-capacity.html]: What are these empty divs for?

File: ui/vic-ui-h5c/vic/src/vic-webapp/src/app/create-vch-wizard/compute-capacity/compute-capacity.html [hunk 2]
```diff
@@ -1,136 +1,348 @@
...
+               [class.invalid]="form.get('memoryLimit').invalid && (form.get('memoryLimit').dirty || form.get('memoryLimit').touched)">
+
+          <input id="memory-limit-selector"
+                 class="form-control"
+                 type="text"
+                 formControlName="memoryLimit">
+          <span class="tooltip-content">{{ formErrMessage }}</span>
+
+          <small class="text-muted">
+            Max: {{ resourceLimits.memory.maxUsage || '-' }} MB, Min: 1 MB
```
Comment (jak-atx) [inline ui/vic-ui-h5c/vic/src/vic-webapp/src/app/create-vch-wizard/compute-capacity/compute-capacity.html]: The Min value shouldn't be defined in the markup it should be same as Max. I know it was already here but if you can update it that would be great.

File: ui/vic-ui-h5c/vic/src/vic-webapp/src/app/create-vch-wizard/compute-capacity/compute-capacity.html [hunk 3]
```diff
@@ -1,136 +1,348 @@
...
+               [class.invalid]="form.get('cpuReservation').invalid && (form.get('cpuReservation').dirty || form.get('cpuReservation').touched)">
+
+          <input id="cpu-reservation-selector"
+                 class="form-control"
+                 type="text"
+                 formControlName="cpuReservation">
+          <span class="tooltip-content">{{ formErrMessage }}</span>
+
+          <small class="text-muted">
+            Max: {{ resourceLimits.cpu.unreservedForPool || '-' }} MHz, Min: 1 MHz
```
Comment (jak-atx) [inline ui/vic-ui-h5c/vic/src/vic-webapp/src/app/create-vch-wizard/compute-capacity/compute-capacity.html]: Same comment as previous. Min value needs to not be defined in template.

File: ui/vic-ui-h5c/vic/src/vic-webapp/src/app/create-vch-wizard/compute-capacity/compute-capacity.html [hunk 4]
```diff
@@ -1,136 +1,348 @@
...
+               [class.invalid]="form.get('memoryReservation').invalid && (form.get('memoryReservation').dirty || form.get('memoryReservation').touched)">
+
+          <input id="memory-reservation-selector"
+                 class="form-control"
+                 type="text"
+                 formControlName="memoryReservation">
+          <span class="tooltip-content">{{ formErrMessage }}</span>
+
+          <small class="text-muted">
+            Max: {{ resourceLimits.memory.unreservedForPool || '-' }} MB, Min: 1 MB
```
Comment (jak-atx) [inline ui/vic-ui-h5c/vic/src/vic-webapp/src/app/create-vch-wizard/compute-capacity/compute-capacity.html]: Min value needs to not be defined in template.

File: ui/vic-ui-h5c/vic/src/vic-webapp/src/app/create-vch-wizard/compute-capacity/compute-capacity.html [hunk 5]
```diff
@@ -1,136 +1,348 @@
...
+               [class.invalid]="form.get('endpointMemory').invalid && (form.get('endpointMemory').dirty || form.get('endpointMemory').touched)">
+
+          <input id="vm-memory-selector"
+                 class="form-control"
+                 type="text"
+                 formControlName="endpointMemory">
+          <span class="tooltip-content">{{ formErrMessage }}</span>
+
+          <small class="text-muted">
+            Max: {{ resourceLimits.memory.maxUsage || '-' }} MB, Min: 1 MB
```
Comment (jak-atx) [inline ui/vic-ui-h5c/vic/src/vic-webapp/src/app/create-vch-wizard/compute-capacity/compute-capacity.html]: Min value needs to not be defined in template.

File: ui/vic-ui-h5c/vic/src/vic-webapp/src/app/create-vch-wizard/create-vch-wizard.component.html [hunk 6]
```diff
@@ -1,7 +1,18 @@
 <!-- inline wizard: https://github.com/vmware/clarity/blob/master/src/app/wizard/wizard-inline.demo.html -->
-<clr-wizard #wizardlg clrWizardSize="lg" class="clr-wizard--inline clr-wizard--no-shadow" style="height: 300px" [clrWizardPreventDefaultCancel]="true"
-  clrWizardPreventModalAnimation="true" (clrWizardCurrentPageChanged)="resizeToParentFrame()" [clrWizardClosable]="false">
-  <clr-wizard-title>Create<br/>Virtual Container Host</clr-wizard-title>
+<clr-wizard #wizardlg
+            clrWizardSize="lg"
+            class="clr-wizard--inline clr-wizard--no-shadow"
+            style="height: 300px"
```
Comment (jak-atx) [inline ui/vic-ui-h5c/vic/src/vic-webapp/src/app/create-vch-wizard/create-vch-wizard.component.html]: Is there any where we can add this into a css class?

File: ui/vic-ui-h5c/vic/src/vic-webapp/src/app/create-vch-wizard/name/vch-creation-wizard-name.html [hunk 7]
```diff
@@ -1,16 +1,48 @@
...
+
+        <label for="nameInput"
+               aria-haspopup="true"
+               role="tooltip"
+               class="form-control tooltip tooltip-validation tooltip-lg tooltip-top-left"
+               [class.invalid]="form.get('name').invalid && (form.get('name').dirty || form.get('name').touched)">
+
+          <input id="nameInput"
+                 class="form-control"
+                 size="50"
```
Comment (jak-atx) [inline ui/vic-ui-h5c/vic/src/vic-webapp/src/app/create-vch-wizard/name/vch-creation-wizard-name.html]: This seems rather arbitrary. Using 'size' can be inconsistent across browsers. Maybe we can use a CSS equivalent size such as 'ch'. Add to style sheet if feasible. But either way let's validate against the mockup and check cross browser.

File: ui/vic-ui-h5c/vic/src/vic-webapp/src/app/create-vch-wizard/name/vch-creation-wizard-name.scss [hunk 8]
```diff
@@ -1,14 +1,4 @@
...
-  }
-  & span.red-text {
-    color: red;
-    margin: 0;
-  }
-}
-
-.smaller-p-body-text {
-  font-size: 12px;
+:host /deep/ clr-signpost .signpost-action {
```
Comment (jak-atx) [inline ui/vic-ui-h5c/vic/src/vic-webapp/src/app/create-vch-wizard/name/vch-creation-wizard-name.scss]: let's verify current browser support for /deep/ selector

File: ui/vic-ui-h5c/vic/src/vic-webapp/src/app/create-vch-wizard/security/security.scss [hunk 9]
```diff
@@ -1,60 +1,9 @@
-div.form-group {
-  & span.red-text {
-    color: red;
-    margin: 0;
-  }
+:host /deep/ clr-signpost .signpost-action {
```
Comment (jak-atx) [inline ui/vic-ui-h5c/vic/src/vic-webapp/src/app/create-vch-wizard/security/security.scss]: verify current browser support for /deep/ selector

**Review by jak-atx — COMMENTED** [review]
Great cleanup, thanks. Just a few small things noted in the comments. And please make sure to test cross browsers and list all browsers/platforms tested in commit msg. Thanks.

File: ui/vic-ui-h5c/vic/src/vic-webapp/src/app/create-vch-wizard/name/vch-creation-wizard-name.scss (hunk 8 above)
Comment (cristianfalcone) [reply ui/vic-ui-h5c/vic/src/vic-webapp/src/app/create-vch-wizard/name/vch-creation-wizard-name.scss]: /deep/ is interpreted by angular cli in build time and rendered with standard css selectors. It allows us to by-pass child components styles encapsulation. Clarity is adding margins to all buttons that are form child elements. Even buttons rendered by Clarity's Signpost components. That's why we need this. I will replace this with its alias (::ng-deep) to make it more specific.

File: ui/vic-ui-h5c/vic/src/vic-webapp/src/app/create-vch-wizard/name/vch-creation-wizard-name.html (hunk 7 above)
Comment (cristianfalcone) [reply ui/vic-ui-h5c/vic/src/vic-webapp/src/app/create-vch-wizard/name/vch-creation-wizard-name.html]: Sorry. This shouldn't be here. The input's parent with .form-group and .row classes, and input .form-control class make the input width adjust to the grid.

File: ui/vic-ui-h5c/vic/src/vic-webapp/src/app/create-vch-wizard/storage-capacity/storage-capacity.html (hunk 1 above)
Comment (cristianfalcone) [reply ui/vic-ui-h5c/vic/src/vic-webapp/src/app/create-vch-wizard/storage-capacity/storage-capacity.html]: This is to set space for previous auto layout columns, so form controls align to the same column. I will add a markup comment per occurrence to explain this.

**Comment by cristianfalcone** [comment]
@jak-atx, the wizard modal size is broken in IE11, I think @jooskim is taking care of this in another issue, but the steps layout looks good in IE11 too.

**Comment by jooskim** [comment]
Thanks @cristianfalcone, I'm addressing this bug in issue #28

**Review by jak-atx — APPROVED** [review]
Looks good. Be sure to rebase fix any conflicts before merge.

**Comment by cristianfalcone** [comment]
@jak-atx Done with rebase. Which strategy should I choose to merge?

**Comment by jak-atx** [comment]
@cristianfalcone once rebased go ahead and merge using 'squash and merge'

**Pull request merged**